lib/ext/heroku/command/addons.rb

```diff
@@ -341,25 +344,27 @@ def destroy
       ).body.keep_if do |attachment|
         attachment['addon']['name'] == addon
       end
-      config_vars = addon_attachments.map { |attachment| attachment['name'] }
 
-      config_vars.each do |var_name|
-        action("Removing #{addon} as #{var_name} from #{app}") {}
-        action("Unsetting #{var_name} and restarting #{app}") {}
+      attachment_names = addon_attachments.map { |attachment| attachment['name'] }
+      attachment_names.each do |attachment_name|
+        action("Removing #{addon} as #{attachment_name} from #{app}") {}
```

Comment:
I think it should be part of the output after the request. We are outputting:

* `Removing :addon as :attachment from :app` once per attachment
* `Unsetting :var1, :var2, ... :varN and restarting :app` for each attached app

One happens before the deletion occurs and the other after but they both say a similar thing. I think they should be consolidated somewhat and both happen as part of an `action("De-provisioning/destroying/whatever :addon") { }` which makes the de-provision call.

The problem, however, is that the `action() {}` thing sucks if you output anything because it goes between the action message and the action result. Ideally it would capture output and replay it after the "... failed"/"... done" happens.